File: backend/app/schemas/terminology.py
from typing import Optional
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class TerminologyCreate(BaseModel):
    term: str = Field(..., max_length=100)
    category: Optional[str] = Field(default=None, max_length=50)
    description: Optional[str] = None


class TerminologyUpdate(BaseModel):
    term: Optional[str] = Field(default=None, max_length=100)
    category: Optional[str] = Field(default=None, max_length=50)
    description: Optional[str] = None


class TerminologyResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    term: str
    category: Optional[str]
    description: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/terminology.ts
import client from './client';

export interface Terminology {
  id: string;
  project_id: string;
  term: string;
  category: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface TerminologyCreate {
  term: string;
  category?: string;
  description?: string;
}

export interface TerminologyUpdate {
  term?: string;
  category?: string;
  description?: string;
}

export const terminologyApi = {
  list: (projectId: string) =>
    client.get<Terminology[]>(`/projects/${projectId}/terminology`),

  create: (projectId: string, data: TerminologyCreate) =>
    client.post<Terminology>(`/projects/${projectId}/terminology`, data),

  update: (id: string, data: TerminologyUpdate) =>
    client.put<Terminology>(`/terminology/${id}`, data),

  delete: (id: string) => client.delete(`/terminology/${id}`),
};
